src/ui/public/query_bar/components/query_bar.tsx

```diff
@@ -533,30 +618,74 @@ export class QueryBarUI extends Component<Props, State> {
             </div>
           </EuiOutsideClickDetector>
         </EuiFlexItem>
-        <EuiFlexItem grow={false}>
-          <EuiButton
-            aria-label={this.props.intl.formatMessage({
-              id: 'common.ui.queryBar.searchButtonAriaLabel',
-              defaultMessage: 'Search',
-            })}
-            data-test-subj="querySubmitButton"
-            color={this.isDirty() ? 'secondary' : 'primary'}
-            fill
-            onClick={this.onClickSubmitButton}
-          >
-            {this.isDirty() ? (
-              <FormattedMessage id="common.ui.queryBar.updateButtonLabel" defaultMessage="Update" />
-            ) : (
-              <FormattedMessage
-                id="common.ui.queryBar.refreshButtonLabel"
-                defaultMessage="Refresh"
-              />
-            )}
-          </EuiButton>
-        </EuiFlexItem>
+        <EuiFlexItem grow={false}>{this.renderUpdateButton()}</EuiFlexItem>
       </EuiFlexGroup>
     );
   }
+
+  private renderUpdateButton() {
+    const button = (
+      <EuiSuperUpdateButton
+        needsUpdate={this.isDirty()}
+        isDisabled={this.state.isDateRangeInvalid}
+        onClick={this.onClickSubmitButton}
+        data-test-subj="querySubmitButton"
+      />
+    );
+    if (this.props.showDatePicker) {
+      return (
+        <EuiFlexGroup responsive={false} gutterSize="s">
+          {this.renderDatePicker()}
+          <EuiFlexItem grow={false}>{button}</EuiFlexItem>
+        </EuiFlexGroup>
+      );
+    } else {
+      return button;
+    }
+  }
+
+  private renderDatePicker() {
+    if (!this.props.showDatePicker) {
+      return null;
+    }
+
+    const recentlyUsedRanges = timeHistory
+      .get()
+      .map(({ from, to }: { from: string; to: string }) => {
```

Comment:
If you declared the `time_history.d.ts` above then you wouldn't need to specify the `: string` portion here

Reply:
I tried removing the `: string;` portion with `time_history.d.ts` and still got TS errors